src/components/modals/WalletListModal2.js

```diff
@@ -1,25 +1,28 @@
 // @flow
 
 import { FormField, MaterialInputStyle } from 'edge-components'
+import type { EdgeCurrencyInfo } from 'edge-core-js'
 import React, { Component, Fragment } from 'react'
 import { FlatList, View } from 'react-native'
 
 import { CryptoExchangeCreateWalletRow } from '../../components/common/CryptoExchangeCreateWalletRow.js'
 import { CryptoExchangeWalletListTokenRowConnected as CryptoExchangeWalletListRow } from '../../connectors/components/CryptoExchangeWalletListRowConnector.js'
+import { type SupportedWalletTypes, getSupportedWalletTypesByCurencyInfos } from '../../modules/Settings/selectors.js'
 import type { GuiWallet, MostRecentWallet } from '../../types/types.js'
 import { scale } from '../../util/scaling.js'
+import { type TokenSelectObject } from '../common/CryptoExchangeWalletListTokenRow.js'
 import { type AirshipBridge, AirshipModal } from './modalParts.js'
 
 export type StateProps = {
+  wallets: { [string]: GuiWallet },
   activeWalletIds: Array<string>,
-  mostRecentWallets: Array<MostRecentWallet>
+  mostRecentWallets: Array<MostRecentWallet>,
+  allCurrencyInfos: Array<EdgeCurrencyInfo>
 }
 
 type OwnProps = {
-  bridge: AirshipBridge<GuiWallet | Object | null>,
+  bridge: AirshipBridge<GuiWallet | TokenSelectObject | SupportedWalletTypes | null>,
```

Comment:
Thanks, this looks much, much better. However, we have a `GuiWalletType` in src/util/CurrencyInfoHelpers.js which is the exact same as your new `SupportedWalletTypes` type, so please use the existing one instead.